.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,69 @@
+name: Benchmark
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - 1.*
+      - 2.*
+  pull_request:
+    branches:
+      - main
+      - 1.*
+      - 2.*
+env:
+  JAVA_VERSION: 21
+  OPENSEARCH_VERSION: 3.0.0
+  OPENSEARCH_ADMIN_PASSWORD: Hello_World123
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+    name: Benchmark
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ env.JAVA_VERSION }}
+      - name: Checkout Branch
+        uses: actions/checkout@v4
+      - name: Set up gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          cache-disabled: false
+      - name: Assemble target plugin
+        run: ./gradlew assemble
+      - name: Prepare Security Plugin
+        run: mv ./build/distributions/opensearch-security-*.zip ./benchmark/docker/
+      - name: Build Docker Image
+        run: docker build --platform linux/amd64 --build-arg VERSION=${{ env.OPENSEARCH_VERSION }} -f benchmark/docker/benchmark.dockerfile -t opensearchproject/security-benchmark:latest benchmark/docker/
+      - name: Run Benchmark Cluster
```

Comment:
I don't think test will produce consistent metrics when run in a cluster.  If we are trying to detect 'large' issues, a cluster adds more variables.  What results have you seen in your testing - can you help build my confidence on these choices?